src/data/finsupp/basic.lean

```diff
@@ -1281,6 +1281,29 @@ lemma multiset_sum_sum [has_zero M] [add_comm_monoid N] {f : α →₀ M} {h : 
   multiset.sum (f.sum h) = f.sum (λa b, multiset.sum (h a b)) :=
 (multiset.sum_add_monoid_hom : multiset N →+ N).map_sum _ f.support
 
+section disjoint_prod_add
+variables [add_comm_monoid M]
+lemma disjoint_prod_add_aux {f1 f2 : α →₀ M} (hd : disjoint f1.support f2.support)
+  {β : Type*} [comm_monoid β] (g : α → M → β) :
+(∏ (x : α) in f1.support, g x (f1 x + f2 x)) = f1.prod g :=
+begin
+  unfold finsupp.prod,
+  rw finset.prod_congr rfl,
+  intros x hx,
+  simp only [not_mem_support_iff.mp (finset.disjoint_left.mp hd hx), add_zero],
+end
```

Comment:
Can you inline this into the other proof with a `have` or `suffices`? The statement seems too weird to justify a global lemma.